src/Microsoft.DotNet.Interactive.ExtensionLab/System.Collections.Generic/EnumerableExtensions.cs

```diff
@@ -2,22 +2,37 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Linq;
+using Microsoft.DotNet.Interactive;
 using Microsoft.DotNet.Interactive.ExtensionLab;
 using Microsoft.DotNet.Interactive.Formatting;
 
 namespace System.Collections.Generic
 {
     public static class EnumerableExtensions
     {
-        public static SandDanceExplorer ExploreWithSandDance<T>(this IEnumerable<T> source)
+        public static SandDanceDataExplorer ExploreWithSandDance<T>(this IEnumerable<T> source, bool immediateDisplay = true, KernelInvocationContext invocationContext = null)
         {
-            var explorer = new SandDanceExplorer();
-            explorer.LoadData(source);
+            var explorer = new SandDanceDataExplorer(source.ToTabularDataResource());
+            if (immediateDisplay)
```

Comment:
This case looks like it should be a different method, or even a simple constructor.

Reply:
Thank you! I missed the point here!